zulip_bots/zulip_bots/bots/beeminder/beeminder.py

```diff
@@ -0,0 +1,100 @@
+import requests
+import logging
+import json
+from typing import Dict, Any, List
+from requests.exceptions import HTTPError, ConnectionError
+
+help_message = '''This is the Beeminder bot. \
+With this, you can add datapoints towards your beeminder goals \
+without having to leave zulip.\n \
+\nSYNTAX - @mention-botname daystamp, value, comment\
+\n* daystamp -> Enter date in this manner: yyyymmdd \
+[NOTE: optional field, default -> current daystamp],\
+\n* value -> Enter a value [NOTE: required field, can be any number],\
+\n* comment -> Add a comment [NOTE: optional field, default -> None]\
+'''
+
+def get_beeminder_response(message_content: str, config_info: Dict[str, str]) -> str:
+    username = config_info['username']
+    goalname = config_info['goalname']
+    auth_token = config_info['auth_token']
+
+    message_content = message_content.strip()
+    if message_content == '' or message_content == 'help':
+        return help_message
+
+    url = "https://www.beeminder.com/api/v1/users/{}/goals/{}/datapoints.json".format(username, goalname)
+    message_pieces = message_content.split(',')
+    for i in range(len(message_pieces)):
+        message_pieces[i] = message_pieces[i].strip()
+
+    if (len(message_pieces) == 1):
+        payload = {
+            "value": message_pieces[0],
+            "auth_token": auth_token
+        }
+    elif (len(message_pieces) == 2):
+        if (message_pieces[1].isdigit()):
+            payload = {
+                "daystamp": message_pieces[0],
+                "value": message_pieces[1],
+                "auth_token": auth_token
+            }
+        else:
+            payload = {
+                "value": message_pieces[0],
+                "comment": message_pieces[1],
+                "auth_token": auth_token
+            }
+    elif (len(message_pieces) == 3):
+        payload = {
+            "daystamp": message_pieces[0],
+            "value": message_pieces[1],
+            "comment": message_pieces[2],
+            "auth_token": auth_token
+        }
+    elif (len(message_pieces) > 3):
+        return "Make sure you follow the syntax.\n You can take a look \
+at syntax by: @mention-botname help"
+
+    try:
+        r = requests.post(url, json=payload)
+
+        if r.status_code != 200:
+            if r.status_code == 401:   # Handles case of invalid key and missing key
+                return "Error. Check your key!"
+            else:
+                return "Error occured : {}".format(r.status_code)   # Occures in case of unprocessable entity
+        else:
+            return "Datapoint created!"   # Handles the case of successful datapoint creation
```

Comment:
It might be helpful to return the link to the datapoint constructed, so that person can then directly click it to view it in beeminder.

Reply:
Hmm...this will be helpful, we can show them a link to their goal ....thanks Rohit 👍